### PR TITLE
enh: Refresh flow when scopes change since last authorization

### LINKED ISSUE #483: Bug: Scope validation uses stale file metadata instead of actual token scopes
## Problem

`get_credentials()` in `auth/google_auth.py` validates required scopes against `credentials.scopes`, which is populated from the `scopes` field in the on-disk credential JSON file. This field is written once at initial authorization time and never updated. It doesn't reflect the actual scopes the access token carries.

This causes valid tokens to be rejected when:

1. The credential file was created with a subset of scopes, but the token was later refreshed (scopeless refresh returns all originally-granted scopes)
2. The server's tool configuration changes (e.g., new tools enabled via `TOOL_TIER`) but existing credentials weren't re-authorized
3. Google's scope hierarchy means a granted scope covers a required one (e.g., `gmail.modify` grants read access, but the literal string `gmail.readonly` is missing from the metadata)

## Scope checks affected

There are 4 separate checks in the codebase, all validating against the same stale source:

| Location | Path | Mode |
|----------|------|------|
| `auth/google_auth.py:591` | OAuth 2.1 credentials | `return None` |
| `auth/google_auth.py:713` | Legacy / single-user | `return None` |
| `auth/service_decorator.py:277` | OAuth 2.1 service auth | `raise GoogleAuthenticationError` |
| `auth/service_decorator.py:307` | OAuth 2.1 session store | `raise GoogleAuthenticationError` |

## Ordering issue

In `google_auth.py`, the scope check at line 713 runs before the token refresh at line 728. This means if scopes don't match the metadata, the function returns `None` immediately and never attempts to refresh even though the refresh would succeed and the token would work.

```python
# Line 713: scope check runs first
if not all(scope in credentials.scopes for scope in required_scopes):
    return None  # Exits here, never reaches refresh below

# Line 728: refresh never reached if scopes mismatch
elif credentials.expired and credentials.refresh_token:
    credentials.refresh(Request())
```

## google-auth library doesn't update scopes on refresh

Even when the refresh path IS reached (token expired but scopes happen to match), `credentials.refresh(Request())` only updates `token` and `expiry`. It does not update `scopes` or `granted_scopes` These are treated as immutable by the library ([docs](https://googleapis.dev/python/google-auth/latest/reference/google.oauth2.credentials.html)). So `credential_store.store_credential()` at line 746 persists the same stale scopes back to disk.

## Reproduction

1. Authorize with a scope set that includes `gmail.modify` but not `gmail.readonly`
2. Store credentials to `~/.google_workspace_mcp/credentials/<email>.json`
3. Call `search_gmail_messages` — it requires `gmail.readonly`
4. The check at line 713 fails because `gmail.readonly` is not in the file's `scopes` list, even though:
   - The access token works fine for Gmail read operations (since `gmail.modify` is a superset of `gmail.readonly` at Google's API level)
   - A token refresh would succeed

## Suggested fix

Google's token refresh response includes a `scope` field listing the actual granted scopes. After refreshing a token, the server should update `credentials.scopes` from this authoritative source before persisting to the credential store.

Since the google-auth library doesn't do this automatically, the refresh path in `get_credentials()` would need to explicitly read the granted scopes from the refresh response and update the credentials object before calling `credential_store.store_credential()`.

Additionally, the scope check should run after the refresh attempt (or at least retry after refresh), so that stale metadata doesn't prevent a valid refresh from happening.

## Related

- #290 — similar class of bug (unnecessary `client_secrets_path` gate blocking token refresh)
- #478 — refresh token not auto-refreshing

## Environment

- Version: v1.11.5
- Mode: Single-user, streamable-http transport
- Containerized deployment

### PR DESCRIPTION
**Pull request by taylorwilsdon**
Closes #483 

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved credential scope validation to treat broader permissions as covering narrower ones, perform scope checks after credential refresh/storage, and add clearer warnings — resulting in more reliable authentication and fewer false missing-scope errors.

* **Tests**
  * Added a comprehensive test suite validating hierarchical scope logic across many edge cases, combinations, and empty/None scenarios.

### CODE REVIEW

File: tests/test_scopes.py [hunk 1]
```diff
@@ -12,10 +12,23 @@
 sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "..")))
 
 from auth.scopes import (
+    CALENDAR_READONLY_SCOPE,
+    CALENDAR_SCOPE,
```
[the remark this answers is not in the document]
Comment (taylorwilsdon) [reply tests/test_scopes.py:16]: Literal hallucination copilot--
